Source/WebCore/css/parser/CSSParserImpl.cpp

```diff
@@ -592,6 +602,138 @@ RefPtr<StyleRuleFontFace> CSSParserImpl::consumeFontFaceRule(CSSParserTokenRange
     return StyleRuleFontFace::create(createStyleProperties(m_parsedProperties, m_context.mode));
 }
 
+static std::pair<FontFeatureValuesType, unsigned> fontFeatureValuesTypeMappings(CSSAtRuleID id)
+{
+    switch (id) {
+    // Styleset has no maximum in theory, but in practise we don't consider more than 99 values.
+    case CSSAtRuleStyleset: return { FontFeatureValuesType::Styleset, 99 }; 
+    case CSSAtRuleStylistic: return { FontFeatureValuesType::Stylistic, 1 };
+    case CSSAtRuleCharacterVariant: return { FontFeatureValuesType::CharacterVariant, 2 };
+    case CSSAtRuleSwash: return { FontFeatureValuesType::Swash, 1 };
+    case CSSAtRuleOrnaments: return { FontFeatureValuesType::Ornaments, 1 };
+    case CSSAtRuleAnnotation: return { FontFeatureValuesType::Annotation, 1 };
+    default:
+        ASSERT_NOT_REACHED();
+        return { };
+    }
+}
+
+RefPtr<StyleRuleFontFeatureValuesBlock> CSSParserImpl::consumeFontFeatureValuesRuleBlock(CSSAtRuleID id, CSSParserTokenRange prelude, CSSParserTokenRange range)
+{
+    // @swash { foo: 3; }
+
+    // Prelude should be empty.
```

Comment:
Does this comment answer "why" rather than "what"?

Reply:
I think so actually 😅  For me, "What?" answer would be "we return early if the prelude is at end" (which indeed is a fairly useless comment).

Which comment would you suggest ?

Reply:
🆗